utils/default.nix

```diff
@@ -50,7 +56,7 @@ let
     then builtins.throw "You need to pass a \"flash-command\" to \"utils-factory\""
     else
       pkgs.writeShellScriptBin "flasher" ''
-        HEX_FILE=${hex}/${keyboard-name}_${keymap-name}.hex
+        HEX_FILE=$(find ${hex}/ -type f -name "*.hex" | head -n 1)
```

Comment:
`find` and `head` are always in the `PATH` of scripts created with `writeShellScriptBin`?
I think so but I'm not 100% sure.

Reply:
I have not yet found out the proper principled way of testing hermeticity when bashing with Nix. So I share this doubt with you.

Reply:
Builds are sandboxed, in this case `pkgs.writeShellScriptBin ...` is a derivation which is built in a sandbox then, **during the build**, it cannot access impurities on your system (e.g. your system's `find` binary). But then you execute this script normally when you run `nix run`. And it will potentially contains reference both to things in the store and things on your system.

Probably it's enough running `PATH="" nix run .#flasher` to check if the generated script contains references to your system.

Reply:
They way it is now, `.#flasher` cannot be the target of `nix run` or `nix build`. In any case, for future reference, the PATH should be set as `PATH="$(dirname $(which nix)):$(dirname $(which git))"` to test this idea.